language_detector/main.py

```diff
@@ -1,9 +1,28 @@
-"""This is the entry point of the program."""
-
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text_lang = "Unknown"
+    a_list = text.split()
+    
+    # For each language, check every word and see if it can be found in the
+    # languages.py file, if so add that word to a list.
+    # Finally, get the length of each of these lists.
+    try:
+        eng = [x for x in a_list for y in languages[2]['common_words'] if x==y]
+        length_english = len(eng)
+    except IndexError:
+        length_english = 0
+    ger = [x for x in a_list for y in languages[1]['common_words'] if x==y]
+    length_german = len(ger)
+    spa = [x for x in a_list for y in languages[0]['common_words'] if x==y]
+    length_spanish = len(spa)
+
+    # Check to see which language had the most words in the text
```

Comment:
You could have simplified this part if you used a for loop to check all the languages, and checked as you went along the loop. Sudo code example:
```
current_best_language = 'Unknown'
current_best_word_match = 0
for language in all_the_languages:
  current_language_word_matches = len([LIST COMPREHENSION FOR WORD MATCHES])
  if current_language_word_matches > current_best_word_match:
    current_best_word_match = current_language_word_matches
    current_best_language = language
```